auction-server/src/auction.rs

```diff
@@ -1066,3 +1034,353 @@ async fn simulate_bid_svm(chain_store: &ChainStoreSvm, bid: &BidSvm) -> Result<(
+
+    #[tracing::instrument(skip_all)]
+    async fn submit_bids(
+        &self,
+        _permission_key: Bytes,
+        bids: Vec<Self::SimulatedBid>,
+        store: Arc<Store>,
+    ) -> Result<Vec<u8>> {
+        let relayer = store.express_relay_svm.relayer.clone();
+        let mut bid = bids[0].clone();
+        let serialized_message = bid.transaction.message.serialize();
+        let relayer_signature_pos = bid
+            .transaction
+            .message
+            .static_account_keys()
+            .iter()
+            .position(|p| p.eq(&relayer.pubkey()))
+            .expect("Relayer not found in static account keys");
+        bid.transaction.signatures[relayer_signature_pos] =
+            relayer.sign_message(&serialized_message);
+        match self.client.send_transaction(&bid.transaction).await {
+            Ok(response) => Ok(response.as_ref().to_vec()),
+            Err(e) => {
+                tracing::error!("Error while submitting bid: {:?}", e);
+                Err(anyhow!(e))
+            }
+        }
+    }
+
+    async fn get_bid_results(
+        &self,
+        bids: Vec<Self::SimulatedBid>,
+        tx_hash: Vec<u8>,
+    ) -> Result<Option<Vec<BidStatus>>> {
+        if bids.len() != 1 {
+            return Err(anyhow!("Invalid number of bids: {}", bids.len()));
+        }
+
+        let status = self
+            .client
+            .get_signature_status_with_commitment(
+                &SignatureSvm::try_from(tx_hash.clone())
+                    .map_err(|e| anyhow!("Invalid svm signature: {:?}", e))?,
+                CommitmentConfig::confirmed(),
+            )
+            .await?;
+
+        match status {
+            Some(res) => Ok(Some(vec![match res {
+                Ok(()) => BidStatus::Won {
+                    index:  0,
+                    result: tx_hash,
+                },
+                Err(_) => BidStatus::Lost {
+                    index:  Some(0),
+                    result: Some(tx_hash),
+                },
+            }])),
+            None => {
+                // not yet confirmed
+                Ok(None)
+            }
+        }
+    }
+}
+
+async fn run_submission_loop<T: Auction>(
+    store: Arc<Store>,
+    chain_store: T,
+    chain_id: String,
+) -> Result<()> {
+    tracing::info!(chain_id = chain_id, "Starting transaction submitter...");
+    let mut exit_check_interval = tokio::time::interval(EXIT_CHECK_INTERVAL);
+
+    let ws_client = chain_store.get_ws_client().await?;
+    let mut stream = T::get_block_stream(&ws_client).await?;
+
+    while !SHOULD_EXIT.load(Ordering::Acquire) {
+        tokio::select! {
+            block = stream.next() => {
+                if block.is_none() {
+                    return Err(anyhow!("Block stream ended for chain: {}", chain_id));
+                }
+
+                tracing::debug!("New block received for {} at {}: {:?}", chain_id, OffsetDateTime::now_utc(), block);
+                store.task_tracker.spawn(
```

Comment:
the frequency of new auctions may differ for solana vs. most evm chains. e.g. we may not want to run a solana auction so frequently to give searchers time to make bids, we may decide to run one every 2 slots. what do you think about making this function defined within the Auction trait implementations?